src/monitor.rs

```diff
@@ -292,6 +292,20 @@ async fn mine_and_submit_solution<T>(
 		return
 	}
 
+	if let Err(e) =
+		ensure_no_previous_solution::<T::Solution>(&api, best_head, signer.account_id()).await
```

Comment:
//cc @kianenigma we only checked this when the block was received (i.e, heads subscriptions) and after solution was mined we need to ensure that the account hasn't submitted at the latest state.